### PR TITLE
Add instructions for manual testing to DG

### PR DESCRIPTION
**Pull request by caipng**

### CODE REVIEW

**Review by chaiwanlin — APPROVED** [review]
lgtm for module and group commands, correct behaviour

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -743,52 +726,323 @@ testers are expected to do more *exploratory* testing.
...
+
+2. Test case: With no module selected, `mkasg n/Assignment 3 w/10 s/30`
+
+    Expected: The console displays an error, reminding the user to select a module first.
+
+3. Test case: With a module selected, `mkasg n/Assignment 3`
+
+    Expected: The console displays an error, reminding the user of the correct command format.
+
+4. Test case: With a module selected, which already has assignments totalling 90% weighage, `mkasg n/Assignment Last w/20 s/30`
```
Comment (marcusc55) [inline docs/DeveloperGuide.md:946]:
```suggestion
4. Test case: With a module selected, which already has assignments totalling 90% weightage, `mkasg n/Assignment Last w/20 s/30`
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -743,52 +726,323 @@ testers are expected to do more *exploratory* testing.
...
+
+    Expected: The console displays an error, reminding the user that the total weighage of assignments in a module cannot exceed 100%.
```
Comment (marcusc55) [inline docs/DeveloperGuide.md:948]:
```suggestion
    Expected: The console displays an error, reminding the user that the total weightage of assignments in a module cannot exceed 100%.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -743,52 +726,325 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Test case: With a module selected, with the assignment to edit having an ID of 1, `edasg 1 n/Midterm w/2.5`
+
+    Expected: The assignment with ID 1 is edited to have name Midterm with weighage 2.5
+
+2. Test case: With a module selected, with no assignment of ID 10, `edasg 10 n/test`
+
+    Expected: The console displays an error, reminding the user that there is no such assignment
+
+3. Test case: With a module selected, with all assignment weighage summing to 90% and an existing assignment of ID 1,  `edasg 1 w/20`
```
Comment (marcusc55) [inline docs/DeveloperGuide.md:960]:
```suggestion
3. Test case: With a module selected, with all assignment weightage summing to 90% and an existing assignment of ID 1,  `edasg 1 w/20`
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -743,52 +726,325 @@ testers are expected to do more *exploratory* testing.
...
+
+    Expected: The console displays an error, reminding the user that the total weighage of assignments in a module cannot exceed 100%.
```
Comment (marcusc55) [inline docs/DeveloperGuide.md:962]:
```suggestion
    Expected: The console displays an error, reminding the user that the total weightage of assignments in a module cannot exceed 100%.
```

**Review by FergusMok — APPROVED** [review]
LGTM

**Pull request merged**